crates/hotshot/types/src/vote.rs

```diff
@@ -291,11 +291,14 @@ impl<TYPES: NodeType> LightClientStateUpdateVoteAccumulator<TYPES> {
             error!("Invalid light client state update vote {vote:?}");
             return None;
         }
-        if !<TYPES::StateSignatureKey as LCV3StateSignatureKey>::verify_state_sig(
-            &state_ver_key,
-            &vote.signature,
-            vote.signed_state_digest,
-        ) {
+        // only verify the new state signature if the auth_root is not default, indicating a vote that was cast from an older vote
+        if vote.auth_root != <FixedBytes<32> as Default>::default()
```

Comment:
we should add `auth_root` to the keys of `vote_outcomes`. Otherwise malicious replica will do sth bad here.